stdlib/3/importlib/abc.pyi

```diff
@@ -70,3 +70,27 @@ if sys.version_info >= (3, 7):
         def is_resource(self, name: str) -> bool: ...
         @abstractmethod
         def contents(self) -> Iterator[str]: ...
+
+if sys.version_info >= (3, 9):
+    from typing import Protocol, runtime_checkable
+    @runtime_checkable
+    class Traversable(Protocol):
+        @abstractmethod
+        def iterdir(self) -> Iterator[Traversable]: ...
+        @abstractmethod
+        def read_bytes(self) -> bytes: ...
+        @abstractmethod
+        def read_text(self, encoding: Optional[str] = ...) -> str: ...
+        @abstractmethod
+        def is_dir(self) -> bool: ...
+        @abstractmethod
+        def is_file(self) -> bool: ...
+        @abstractmethod
+        def joinpath(self, child: Traversable) -> Traversable: ...
+        @abstractmethod
+        def __truediv__(self, child: Traversable) -> Traversable: ...
+        @abstractmethod
+        def open(self, mode: str = ..., *args, **kwargs) -> IO: ...
```

Comment:
If I understand the docstring correctly, `mode` would be better annotated as `Literal["r", "rb"]`. Also a nit: `args` and `kwargs` could be annotated as `Any`.